pxtcompiler/emitter/emitter.ts

```diff
@@ -2155,7 +2165,7 @@ ${lbl}: .short 0xffff
             }
 
             // otherwise we assume a lambda
-            if (args.length > 3)
+            if (args.length > (isStackMachine() ? 2 : 3))
                 userError(9217, lf("lambda functions with more than 3 arguments not supported"))
```

Comment:
update error message with number of args?

Reply:
didn't want to mess with localization, also I guess we may want to lift this restriction